File: app/src/main/java/io/github/sithengineer/motoqueiro/data/sync/DummyAccountCreator.java
package io.github.sithengineer.motoqueiro.data.sync;

import android.accounts.Account;
import android.accounts.AccountManager;
import android.app.Activity;
import android.content.Context;
import android.support.annotation.NonNull;

public class DummyAccountCreator {

  private static final String AUTHORITY = "io.github.sithengineer.motoqueiro.data.sync";
  private static final String ACCOUNT_TYPE = "";
  private static final String ACCOUNT = "dummyaccount";

  public void addAccount(@NonNull Context context) {
    AccountManager accountManager =
        (AccountManager) context.getSystemService(Activity.ACCOUNT_SERVICE);
    Account account = new Account(ACCOUNT, ACCOUNT_TYPE);
    if (accountManager.addAccountExplicitly(account, null, null)) {
      // account was added
    } else {
      // account already exists or an error has occurred
    }
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/data/sync/DummyAccountCreator.kt
package io.github.sithengineer.motoqueiro.data.sync

import android.accounts.Account
import android.accounts.AccountManager
import android.app.Activity
import android.content.Context

class DummyAccountCreator {

  fun addAccount(context: Context): Account? {
    val accountManager = context.getSystemService(Activity.ACCOUNT_SERVICE) as AccountManager
    val account = Account(ACCOUNT, ACCOUNT_TYPE)
    return if (accountManager.addAccountExplicitly(account, null, null)) {
      // account was added
      account
    } else {
      // account already exists or an error has occurred
      null
    }
  }

  companion object {
    private const val AUTHORITY = "io.github.sithengineer.motoqueiro.data.sync"
    private const val ACCOUNT_TYPE = ""
    private const val ACCOUNT = "dummyaccount"
  }
}
